### ISSUE TITLE
Tracked object can only be used once. Using same object twice throws exception.

### ISSUE THREAD

**Issue by shqyking** [OfficeDev/office-js#3925]
# Provide required information needed to triage your issue
I need to use Word.Paragraph object across contexts. So I maintained a list. 

Step1: put tracked paragraphs to the list in Word.run(). 

Step2: when user choose certain paragraph, call `paragraph.select()` and `paragraph.context.sync()` to jump to that paragraph. 

However, if same paragraph is selected more than once, error is thrown.

<img width="870" alt="Screenshot 2023-12-15 at 18 05 58" src="https://github.com/OfficeDev/office-js/assets/5377761/852c032a-e548-4d76-9a10-10fbe2ef71f0">

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: PC Desktop & Mac
* Host [Excel, Word, PowerPoint, etc.]:Word
* Office version number: 2311(PC); 16.78(mac)
* Operating System: Win11; MacOs 14.0
* Browser (if using Office on the web): Edge 120(PC); Safari
 
## Expected behavior

A tracked paragraph(or other object) can call `select()`(or other functions) multiple times before it is untracked.

## Current behavior

It throws exception when a tracked paragraph is called `select()` more than once.

## Steps to reproduce

Use the minimum reproducible script here: https://gist.github.com/shqyking/c93a8471b052211766ff2dbc13830fd9.
1. First click "Get tracked object"
2. Then click "Select first paragraph" multiple times.

## Link to live example(s)

https://gist.github.com/shqyking/c93a8471b052211766ff2dbc13830fd9

## Context

I need it to display certain paragraphs in add-in UI, and let user select which of those for further process. User will have the need to navigate to those paragraphs for understand the context.

**Comment by shqyking** [comment]
Any ideas?

**Comment by penglongzhaochina** [comment]
@xuruiyao-msft  could you please take a look?

**Comment by xuruiyao-msft** [comment]
@shqyking Thanks for reaching out to us. Would you please replace getTrackedParagraphs function with the below code, there's a comment explains the reason. Both the parent collection and the child items should be tracked so that the items can be accessible across multiple context :

`async function getTrackedParagraphs() {
  console.log(myParagraphs.length);
  await Word.run(async (context) => {
    if(myParagraphs.length!=0){
      myParagraphs
    }
    const paragraphs = context.document.body.paragraphs;
    paragraphs.load("items");
    paragraphs.track();
    await context.sync();
    /* Would you please try to add this piece of snippet to your code? 
      If you're using this object across .sync calls and outside the sequential execution of a ".run" batch, and get an "InvalidObjectPath" error when setting a property or invoking a method on the object, you need to add the object to the tracked object collection when the object was first created. If this object is part of a collection, you should also track the parent collection.
    */
    **for (let index = 0; index < paragraphs.items.length; index++) {
      paragraphs.items[index].track();
    }
    await context.sync();**

    myParagraphs = paragraphs.items;
  });
}`

**Comment by xuruiyao-msft** [comment]
@shqyking Thanks for reaching out to us. Would you please replace getTrackedParagraphs and selectFirstParagraph function with the below code, there're two comments explain the reason. Both the parent collection and the child items should be tracked so that the items can be accessible across multiple context. And the tracked items should be put as a parameter into Word.run when they're accessed in another context:

````
async function getTrackedParagraphs() {
  console.log(myParagraphs.length);
  await Word.run(async (context) => {
    if(myParagraphs.length!=0){
      myParagraphs
    }
    const paragraphs = context.document.body.paragraphs;
    paragraphs.load("items");
    paragraphs.track();
    await context.sync();
    /* Would you please try to add this piece of snippet to your code?
      If you're using this object across .sync calls and outside the sequential execution of a ".run" batch, and get an 
"InvalidObjectPath" error when setting a property or invoking a method on the object, you need to add the object 
to the tracked object collection when the object was first created. If this object is part of a collection, you should 
also track the parent collection.
    */
    for (let index = 0; index < paragraphs.items.length; index++) {
      paragraphs.items[index].track();
    }
    await context.sync();
    myParagraphs = paragraphs.items;
  });
}

// Would you please add the tracked list variable (myParagraphs) as the first parameter in Word.run?
async function selectFirstParagraph() {
  await Word.run(myParagraphs, async (context) => {
    myParagraphs[0].select();
    await myParagraphs[0].context.sync();
  });
}
`
```

**Comment by shqyking** [comment]
Yes, it works. Looks like I should track both the parent collection as well as the individual paragraph in the collection. 

If Tracking only parent collection is not enough, should make it fail when calling `select()` for the first time? Otherwise it gives user a wrong impression that tracking only parent collection works. If I didn't call `select()` twice, I won't find out the problem.

**Comment by xuruiyao-msft** [comment]
@shqyking That's a good suggestion. I will put the work item to our backlog list. Once there's any progress about the optimization we'll sync with you. Thanks for reaching out to us again.
